word_guess.rb

```diff
@@ -0,0 +1,269 @@
+    @good_prize = ["elephant hotdog", "invisible pineapple", "expensive whiskey", "customized puppet of yourself"].sample
+  end
+
+
+  # this method enables playing the game
+  def play_the_game
+    pick_a_word
+    update_board
+    until @wrong_count == 0
+      guess
+    end
+    lose_game
+  end
+
+  private
+
+  # this method defines the difficulty level and sets the word to guess
+  def pick_a_word
+    if @word == ""
+      puts "What difficulty would you like to play at? There's easy(E), medium(M), or hard(H)!"
+      print "Difficulty: "
+      @difficulty = gets.chomp.upcase
+      if @difficulty == "E" || @difficulty == "EASY"
+        @word = @pick_word[:easy]
+        @word = @word.upcase
+      elsif @difficulty == "M" || @difficulty == "MEDIUM"
+        @word = @pick_word[:medium]
+        @word = @word.upcase
+      elsif @difficulty == "H" || @difficulty == "HARD"
+        @word = @pick_word[:hard]
+        @word = @word.upcase
+      else
+        # if they try to enter anything except a difficulty level
+        # the game throws up and goes home
+        puts "I don't know what you want, please start over!"
+        exit
+      end
+      @word_so_far = "_" * @word.length
+    end
+  end
+
+  # this method operates getting guesses and checking them.
+  def guess
+    get_input
+    check_letter
+    if word_done?
+      win_game
+    end
+    if !more_tries?
+      lose_game
+    end
+  end
+
+  # this method actually gets the input
+  def get_input
+    puts "What letter do you guess?"
+    @guess = gets.chomp.upcase
+
+    # this checks if you guessed the right word.
+    while @guess.length > 1
+      if @guess == @word.upcase
+        win_game
+      else
+        #second time we use this- refactor into a method?
+        @wrong_count -= 1
+        puts "You have made a terrible guess!"
+        puts "How could you pick that letter?!"
+        @used_letters.push(@guess)
+        update_board
+        puts "You have #{ @wrong_count } guesses left."
+        @guess = gets.chomp.upcase
+      end
+    end
+
+    # this checks if the guess is a single character and if that character is a letter!
+    while @guess.length != 1 || @alphabet_array.include?(@guess) == false
+        puts "Your guess is unusable, please guess again!"
+        @guess = gets.chomp.upcase
+    end
+    # echoes guess to user
+    puts "Your guess was #{ @guess }."
+  end
+
+  # this checks if the letter is in the word, and if you already guessed it.
+  def check_letter
+
+    unless @used_letters.include?(@guess) == false
+        puts "You have already guessed this letter!"
+        puts "Please guess again!"
+        @guess = gets.chomp.upcase
+    else
+        # add letter to used letters array
+        @used_letters.push(@guess)
+    end
+
+    if @word.upcase.include?(@guess)
+      for index in 0..@word.length - 1
+        if @word[index].upcase == @guess
```

Comment:
Since the `pick_a_word` method already does the `upcase` on the whole word, you don't need to use this again to `upcase` each individual character.